File: src/cpp/group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        vector<vector<string>> ans;
        if(strs.size() == 0)
            return ans;
        
        unordered_map<string, vector<string>> mp;
        for(string s : strs){
            string temp = s;
            sort(temp.begin(), temp.end());
            mp[temp].push_back(s);
        }
        
        for(auto it: mp){
            ans.push_back(it.second);
        }
        
        return ans;
    }
};

File: src/python/group-anagrams.py
#https://leetcode.com/problems/group-anagrams/
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        d={}
        for words in strs:
            key=tuple(sorted(words))
            d[key]=d.get(key,[])+[words]
        return d.values()